### PR TITLE
Connector

### PR DESCRIPTION
**Pull request by ansahdavid**

### CODE REVIEW

**Review by warrenskiwork1979 — APPROVED** [review]

File: app/models/Registration.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ */
+
+package models
+
+import play.api.libs.json.Json
+import uk.gov.hmrc.domain.Vrn
+
+import java.time.LocalDate
+
+case class Registration(
```
Comment (TheNickWilson) [inline app/models/Registration.scala:24]: This case class looks to be identical to RegistrationRequest - is this needed?

File: app/services/RegistrationService.scala [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        ukVatRegisteredPostcode,
+        vatRegisteredInEu,
+        euVatDetails,
+        businessAddress,
+        businessContactDetails,
+        websites
+      )
+  }
+
+  def fromUserAnswers(answers: UserAnswers): Option[RegistrationRequest] = {
```
Comment (TheNickWilson) [inline app/services/RegistrationService.scala:58]: This method just seems to wrap the other - do we need both?

File: run_all_tests.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
+sbt -mem 2048 scalastyle compile coverage test coverageOff coverageReport
```
Comment (TheNickWilson) [inline run_all_tests.sh:2]: Should we include it:test here too?

File: test/connectors/RegistrationRequestConnectorSpec.scala [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  val registration = RegistrationData.createNewRegistration()
+
+  implicit private lazy val hc: HeaderCarrier = HeaderCarrier()
+
+  private def application: Application =
+    applicationBuilder()
+      .configure("microservice.services.one-stop-shop-registration.port" -> server.port)
+      .build()
+
+  "submitRegistration" - {
```
Comment (TheNickWilson) [inline test/connectors/RegistrationRequestConnectorSpec.scala:38]: Could we add tests for other cases - e.g. when the backend returns an error code or a 409?

File: app/models/Registration.scala (hunk 1 above)
Comment (warrenskiwork1979) [reply app/models/Registration.scala]: Have removed Registration model and using RegistrationRequest instead

File: app/services/RegistrationService.scala (hunk 2 above)
Comment (warrenskiwork1979) [reply app/services/RegistrationService.scala]: Removed wrapper

File: run_all_tests.sh (hunk 3 above)
Comment (warrenskiwork1979) [reply run_all_tests.sh]: @TheNickWilson - Have added it:test but had to remove scalastyle in order for it to run.

File: test/connectors/RegistrationRequestConnectorSpec.scala (hunk 4 above)
Comment (warrenskiwork1979) [reply test/connectors/RegistrationRequestConnectorSpec.scala:37]: Will do this once we've made the changes on the backend so we no what to expect here.

**Review by susan-badel — APPROVED** [review]

**Pull request merged**